### ISSUE TITLE
Grammar action errors need to be more informative

### ISSUE THREAD

**Issue by stoneRdev** [pegjs/pegjs#668]
### Issue type

- **Bug Report:**  _no_
- **Feature Request:** _yes_
- **Question:** _no_
- **Not an issue:** _no_

### Prerequisites

- **Can you reproduce the issue?:** _yes_
- **Did you search the repository issues?:**  _yes_ (kind of)
- **Did you check the forums?:** _no_ (what forums?)
- **Did you perform a web search (google, yahoo, etc)?:** _yes_

### Description

So, as it stands, grammar action errors are about as vague than C++ segfaults. 
For example, take this small grammar:
```
{
	const def1 = 1
        const def2 = 2
	const a = [def1 def2]
}

line = i / l

i = [ \t\n\r+]
l = [a-z]i
```
The error output is: `missing ] after element list`
Now, while this is fine in such a small example (its easy to spot), errors like this in a large grammar file is a headache.

I'd like to request the action errors to be reported the same as grammar errors, for example:
```
line = i / l

i = [ \t\n\r+
l = [a-z]i
```
outputs `Line 3, column 5: Expected "!", "$", "&", "(", ".", character class, comment, end of line, identifier, literal, or whitespace but "[" found.`, which is magnitudes more helpful.

I think a good solution to this would be to at least include the line number for the action errors
### Steps to Reproduce

1.  Make a grammar file
2.  Make a small typo (or other mistake) in javascript action
3.  Observe cryptographically secure error message

**Example code:**
Oops, provided above in description

**Expected behavior:**

I'd expect errors in action contexts to at least be descriptive enough to find the error
**Actual behavior:**

Errors in action context are vague and not helpful
### Software

- **PEG.js:**  ^0.10.0
- **Node.js:** v14.16.0
- **NPM or Yarn:** yarn (primary) 1.22.10, npm 6.14.11
- **Browser:**  N/A
- **OS:**  Windows 8.1 32 bit os, 64 bit processor
- **Editor:** Sublime 3.2.2 all day

**Comment by Mingun** [comment]
Duplicate of #93, and if you want that something will be done, it is better now to report at https://github.com/peggyjs/peggy/issues. This project is dead for many years

**Comment by stoneRdev** [comment]
@Mingun I have seen mention of it, but am hesitant to switch to new libraries like that. Maybe I should try it out

**Comment by Mingun** [comment]
The first released version [1.0.0](https://github.com/peggyjs/peggy/releases/tag/v1.0.0) is just rebranded pegjs of dozens commits ago (but nothing super-important was lost.. maybe only the `@` operator, but it will be [landed soon](https://github.com/peggyjs/peggy/pull/89))

**Comment by stoneRdev** [comment]
@Mingun Sold! haha I'll give it a swap out right now. Backward compatibility is my main worry, but since I dont use the `@` operator,  I don't really see a problem in switching over. I guess I should move this over there as well
